forte/processors/data_augment/algorithms/eda_processors.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Data augmentation processors from the paper "EDA: Easy Data Augmentation
+Techniques for Boosting Performance on Text Classification Tasks", including
+Random Swap, Random Insertion and Random Deletion. All three processors are
+implemented based on the ReplacementDataAugmentProcessor.
+"""
+
+import random
+from typing import List, Tuple, Dict
+from nltk.corpus import stopwords
+from math import ceil
+
+from forte.data.data_pack import DataPack
+from forte.data.multi_pack import MultiPack
+from forte.processors.base import ReplacementDataAugmentProcessor
+from forte.utils.utils import get_class, create_class_with_kwargs
+
+__all__ = [
+    "RandomSwapDataAugmentProcessor",
+    "RandomInsertionDataAugmentProcessor",
+    "RandomDeletionDataAugmentProcessor",
+]
+
+
+class RandomSwapDataAugmentProcessor(ReplacementDataAugmentProcessor):
+    r"""
+    Data augmentation processor for the Random Swap operation.
+    Randomly choose two words in the sentence and swap their positions.
+    Do this n times.
+    """
+
+    def _process(self, input_pack: MultiPack):
+        augment_entry = get_class(self.configs["augment_entry"])
+        new_packs: List[Tuple[str, DataPack]] = []
+
+        for pack_name, data_pack in input_pack.iter_packs():
+            annotations = list(data_pack.get(augment_entry))
+            if len(annotations) > 0:
+                replace_map: Dict = {}
+                for _ in range(ceil(self.configs['alpha'] * len(annotations))):
+                    swap_idx = random.sample(range(len(annotations)), 2)
+                    new_idx_0 = swap_idx[1] if swap_idx[1] not in replace_map \
+                        else replace_map[swap_idx[1]]
+                    new_idx_1 = swap_idx[0] if swap_idx[0] not in replace_map \
+                        else replace_map[swap_idx[0]]
+                    replace_map[swap_idx[0]] = new_idx_0
+                    replace_map[swap_idx[1]] = new_idx_1
+                pid: int = data_pack.pack_id
+                for idx in replace_map:
+                    self.replaced_annos[pid]\
```

Comment:
I think we should let the base augmented to maintain the information such as `replaced_annos`, can we call the `replace` function?